perk-store-services/src/main/java/org/exoplatform/perkstore/service/PerkStoreService.java

```diff
@@ -304,9 +305,12 @@ public List<ProductOrder> getOrders(OrderFilter filter, String username) throws
         if (!canViewProduct(product, username, isPerkStoreManager(username))) {
           throw new PerkStoreException(PRODUCT_ACCESS_DENIED, product.getTitle(), username);
         }
+        isProductOwner = StringUtils.equals(product.getCreator().getId(), username)
+            || StringUtils.equals(product.getReceiverMarchand().getId(), username);
```

Comment:
please define this variable inside the adequate `else` block since it will be used there only
In addition, please add Null Checks as you can see below :
```java
    } else {
      boolean isProductOwner = product != null && ((product.getCreator() != null && StringUtils.equals(product.getCreator().getId(), username))
            || (product.getReceiverMarchand() != null && StringUtils.equals(product.getReceiverMarchand().getId(), username)));
      orders = perkStoreStorage.getOrders(username, filter, isPerkStoreManager, isProductOwner);
    }
```